Sources/ResourceProvider/Cache/AnyAsyncCache.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  AnyAsyncCache.swift
+//  swift-resource-provider
+//
+//  Created by Óscar Morales Vivó on 4/11/23.
+//
+
+/**
+ Type-erased ``AsyncCache``.
+
+ This wrapper value type can be used to build up adapters for actual cache types, build mocks for testing, and makes for
+ a good specific type to use for non-generic logic to store an async cache.
+ */
+public struct AnyAsyncCache<ID: Hashable & Sendable, Value: Sendable> {
+    /**
+     A type-erased cache has its functionality injected as blocks.
+     - Parameters:
+       - valueForID: Block that implements `AsyncCache.value(for:)`
+       - storeValueForID: Block that implements `AsyncCache.store(value:id:)`
+     */
+    public init(
+        valueForID: @escaping @Sendable (ID) async -> Value? = { _ in nil },
+        storeValueForID: @escaping @Sendable (Value, ID) async -> Void = { _, _ in }
+    ) {
+        self.valueForID = valueForID
+        self.storeValueForID = storeValueForID
+    }
```

Comment:
it might be nice to have a convenience initializer that looks like:

```swift
public init<Cache: AsyncCache<ID, Value>>(_ cache: Cache) {
    valueForID = cache.value(for:)
    storeValueForID = cache.store(value:id:)
}
```

here and in the other `Any*` classes

Reply:
I've mostly left that for the `eraseToAnyCache` methods instead.

I would consider this if either there's more demand for it or it helps with `Sendable` ergonomics.